qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py

```diff
@@ -87,21 +89,29 @@ def __init__(
         # encodes the target data if categorical
         self._target_encoder = OneHotEncoder(sparse=False) if one_hot else LabelEncoder()
 
-    def fit(self, X: np.ndarray, y: np.ndarray):  # pylint: disable=invalid-name
+        # For ensuring the number of classes matches those of the previous
+        # batch when training from a warm start.
+        self._num_classes: int | None = None
+
+    @property
+    def num_classes(self) -> int:
+        """The number of classes found in the most recent fit."""
```

Comment:
If I call this before doing 'fit' then it will return None as set in the constructor. We could state that and indicate that in the return type;  or alternatively raise an exception stating that instance has not been fitted. I think something needs to be done here in that regard

Reply:
Good point. I've now documented this in the return type and docstring.